site/docs/plugins/commands.md

```diff
@@ -5,10 +5,628 @@ next: false
+Commands not following this rules can still be registered, used and handled, but will never be displayed on the user menu as such.
+:::
+
+**Be aware** that `setCommands` and `setMyCommands` only affects the commands displayed in the user's commands menu, and not the actual access to them.
+You will learn how to implement restricted command access in the [Scoped Commands](#scoped-commands) section.
+
+### Grouping Commands
+
+Since we can split and group our commands into different instances, it allows for a much more idiomatic command file organization.
+
+Let's say we want to have developer-only commands.
+We can achieve that with the following code structure:
+
+```ascii
+src/
+├─ commands/
+│  ├─ admin.ts
+│  ├─ users/
+│  │  ├─ group.ts
+│  │  ├─ say-hi.ts
+│  │  ├─ say-bye.ts
+│  │  ├─ ...
+├─ bot.ts
+├─ types.ts
+tsconfig.json
+```
+
+The following code group exemplifies how we could implement a developer only command group, and update the Telegram client Command menu accordingly.
+Make sure you take notice of the different patterns being use in the `admin.ts` and `group.ts` file-tabs.
+
+::: code-group
+
+```ts [types.ts]
+export type MyContext = Context & CommandsFlavor<MyContext>;
+```
+
+```ts [bot.ts]
+import { devCommands } from "./commands/admin.ts";
+import { userCommands } from "./commands/users/group.ts";
+import type { MyContext } from "./types.ts";
+
+export const bot = new Bot<MyContext>("MyBotToken");
+
+bot.use(commands());
+
+bot.use(userCommands);
+bot.use(devCommands);
+```
+
+```ts [admin.ts]
+import { userCommands } from './users/group.ts'
+import type { MyContext } from '../types.ts'
+
+export const devCommands = new CommandGroup<MyContext>()
+
+devCommands.command('devlogin', 'Greetings', async (ctx, next) => {
+   if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
+      await ctx.reply('Hi to me')
+      await ctx.setMyCommands(userCommands, devCommands)
+   } else {
+     await next()
+   }
+})
+
+devCommands.command('usercount', 'Greetings', async (ctx, next) => {
+   if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
+      await ctx.reply(
+        `Active users: ${/** Your business logic */}`
+    )
+   } else {
+     await next()
+   }
+})
+
+devCommands.command('devlogout', 'Greetings', async (ctx, next) => {
+    if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
+       await ctx.reply('Bye to me')
+       await ctx.setMyCommands(userCommands)
+   } else {
+     await next()
+   }
+ })
+```
+
+```ts [group.ts]
+import sayHi from "./say-hi.ts";
+import sayBye from "./say-bye.ts";
+import etc from "./another-command.ts";
+import type { MyContext } from "../../types.ts";
+
+export const userCommands = new CommandGroup<MyContext>()
+  .add([sayHi, sayBye]);
+```
+
+```ts [say-hi.ts]
+import type { MyContext } from "../../types.ts";
+
+export default new Command<MyContext>("sayhi", "Greetings", async (ctx) => {
+  await ctx.reply("Hello little User!");
+});
+```
```

Comment:
This snippet returns errors

types.ts
```ts
import type { Context } from "https://deno.land/x/grammy@v1.31.0/mod.ts";
import type { CommandsFlavor } from "https://deno.land/x/grammy_commands@v0.10.0/mod.ts";

export type MyContext = Context & CommandsFlavor<MyContext>;
```
bot.ts
```ts
import { Bot } from "https://deno.land/x/grammy@v1.31.0/mod.ts";
import { devCommands } from "./commands/admin.ts";
import { userCommands } from "./commands/users/group.ts";
import { commands } from "https://deno.land/x/grammy_commands@v0.10.0/mod.ts";
import type { MyContext } from "./types.ts";

export const bot = new Bot<MyContext>(""); // <-- put your bot token between the "" (https://t.me/BotFather)

bot.use(commands());

bot.use(userCommands);
bot.use(devCommands);
```

commands/admin.ts
```
1. Property 'env' does not exist on type 'CommandContext<MyContext>'.
    Property 'env' does not exist on type 'FilteredContext<NarrowMatch<MyContext, string>, Update & Record<"message", Record<"entities" | AddTwins<"message", "entities">, NotUndefined> & Partial<...>> & Partial<...>>'.
  
2. Type 'CommandGroup<MyContext>' has no properties in common with type 'SetBotCommandsOptions'.
```
```ts
import { userCommands } from "./users/group.ts";
import type { MyContext } from "../types.ts";
import { CommandGroup } from "https://deno.land/x/grammy_commands@v0.10.0/mod.ts";

export const devCommands = new CommandGroup<MyContext>();

devCommands.command("devlogin", "Login", async (ctx, next) => {
  if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
    await ctx.reply("Hello, fellow developer! Are we having coffee today too?");
    // Error (2) ----------------------------- v
    await ctx.setMyCommands(userCommands, devCommands);
  } else {
    await next();
  }
});

devCommands.command(
  "usercount",
  "Count the active users",
  async (ctx, next) => {
    if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
      await ctx.reply(`Active users: `);
    } else {
      await next();
    }
  },
);

devCommands.command("devlogout", "Logout", async (ctx, next) => {
  if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
    await ctx.reply("Until next commit!");
    await ctx.setMyCommands(userCommands);
  } else {
    await next();
  }
});
 ```
 
commands/users/group.ts
```
Property 'add' does not exist on type 'CommandGroup<MyContext>'.
```
 ```ts
import sayHi from "./say-hi.ts";
import sayBye from "./say-bye.ts";
import type { MyContext } from "../../types.ts";
import { CommandGroup } from "https://deno.land/x/grammy_commands@v0.10.0/mod.ts";

export const userCommands = new CommandGroup<MyContext>()
  .add([sayHi, sayBye]);
 ```

commands/users/say-hi.ts
```
1. Type '(ctx: any) => Promise<void>' has no properties in common with type 'Partial<CommandOptions>'.
2. Parameter 'ctx' implicitly has an 'any' type.
```
 ```ts
import { Command } from "https://deno.land/x/grammy_commands@v0.10.0/mod.ts";
import type { MyContext } from "../../types.ts";

export default new Command<MyContext>("sayhi", "Say hello", async (ctx) => {
  await ctx.reply("Hello, little user!");
});
```

Reply:
admin.ts errors
1: this error could be fixed by changing the retrieval of `DEVELOPER_ID` from ctx.env, to a placerholder `your developer id here`. If not, setting of `ctx.env.DEVELOPER_ID` should be added to the snippet logic + the type
```
type MyContext = Context &
    CommandsFlavor<MyContext> &
  {
        env: {
            DEVELOPER_ID: number
        }
    }
```
(I left it implicitly before, I think the placeholder it's okay, and let anyone resolve the logic of how to get it by themself?)

2: 
```ts
async (ctx, next) => {
        if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
            await ctx.reply('Hi to me')
            await ctx.setMyCommands([
                userCommands,
                devCommands,
            ]) 
        } else next()
    }
    ```
    
    `groups.ts` and `say-hi.ts` errors should be fixed by caching the plugin again (the `next` branch was supposed to be merged to `main`, but it was not)

Reply:
should be ok now. I removed the unnecessary env